scripts/biomedical/mesh/format_mesh.py

```diff
@@ -0,0 +1,278 @@
+        'DateEstablished-Day', 'ConceptID', 'ConceptName', 'TermID', 'TermName'
+    ])
+    df_2 = (df_2.set_index('DescriptorId').apply(
+        lambda x: x.apply(pd.Series).stack()).reset_index().drop('level_1', 1))
+    df_2['Qualifier_dcid'] = 'bio/' + df_2['QualifierID'].astype(str)
+    df_2['Descriptor_dcid'] = 'bio/' + df_2['DescriptorId'].astype(str)
+    return df_2
+
+
+def format_concept_df(df):
+    """
+    Modifies the original dataframe to retain all
+    descriptor entries/properties only
+    Args:
+        df = pandas df with date columns
+    Returns:
+        df_3 = df with concept properties only
+    """
+    df_3 = df
+    df_3 = df_3.drop(columns=[
+        'DescriptorName', 'DateCreated-Year', 'DateCreated-Month',
+        'DateCreated-Day', 'DateRevised-Year', 'DateRevised-Month',
+        'DateRevised-Day', 'DateEstablished-Year', 'DateEstablished-Month',
+        'DateEstablished-Day', 'QualifierID', 'QualifierName',
+        'QualifierAbbreviation', 'TermID', 'TermName'
+    ])
+    df_3 = df_3 = (df_3.set_index('DescriptorId').apply(
+        lambda x: x.apply(pd.Series).stack()).reset_index().drop('level_1', 1))
+    df_3['Concept_dcid'] = 'bio/' + df_3['ConceptID'].astype(str)
+    df_3['Descriptor_dcid'] = 'bio/' + df_3['DescriptorId'].astype(str)
+    return df_3
+
+
+def format_term_df(df):
+    """
+    Modifies the original dataframe to retain all
+    descriptor entries/properties only
+    Args:
+        df = pandas df with date columns
+    Returns:
+        df_4 = df with term properties only
+    """
+    df_4 = df
+    df_4 = df_4.drop(columns=[
+        'DateCreated-Year', 'DateCreated-Month', 'DateCreated-Day',
+        'DateRevised-Year', 'DateRevised-Month', 'DateRevised-Day',
+        'DateEstablished-Year', 'DateEstablished-Month', 'DateEstablished-Day',
+        'QualifierID', 'QualifierName', 'QualifierAbbreviation',
+        'DescriptorName', 'DescriptorId'
+    ])
+    df_4 = df_4.apply(lambda x: x.explode() if x.name in
+                      ['ConceptID', 'ConceptName', 'TermID', 'TermName'] else x)
+    df_4 = df_4.reset_index(drop=True)
+    df_4 = df_4.apply(lambda x: x.explode()
+                      if x.name in ['TermID', 'TermName'] else x)
+    df_4['Term_dcid'] = 'bio/' + df_4['TermID'].astype(str)
+    df_4['Concept_dcid'] = 'bio/' + df_4['ConceptID'].astype(str)
+    return df_4
+
+
+def mesh_wrapper(file_input):
+    """
+    Takes in the xml file, runs all helper functions
+    and outputs resulting files
+    Args:
+        file_input = path to mesh xml file
+    """
+    df = format_mesh_xml(file_input)
+    df1 = date_modify(df)
+    df1 = format_descriptor_df(df1)
+    df2 = format_qualifier_df(df)
+    df3 = format_concept_df(df)
+    df4 = format_term_df(df)
+
+    df1.to_csv('descriptor_mesh.csv')
+    df2.to_csv('qualifier_mesh.csv')
+    df3.to_csv('concept_mesh.csv')
+    df4.to_csv('term_mesh.csv')
```

Comment:
I wonder if it's more natural if we matched the file name to the TMCF name in tmcf/ directory?

i.e., mesh_descriptor.csv instead of description_mesh.csv

Reply:
renamed to match